### ISSUE TITLE
Some exceptions encountered using mlx.core.dequantize

### ISSUE THREAD

**Issue by Redmept1on** [ml-explore/mlx#550]
**Describe the bug**

**Problem1:** 
When I use mlx.core.dequantize, I cannot adjust the appropriate scales and biases based on the documentation and error information. When I use the following code, I don't know how to adjust my scales and biases to match the default group_size and bits.
```python
import mlx.core as mx

quantized_w = mx.array([[1, 2], [3, 4]],dtype=mx.uint32)
scales = mx.array([[0.5, 0.5], [0.5, 0.5]])
biases = mx.array([[1.0, 1.0], [1.0, 1.0]])

group_size = 64
bits = 4

dequantized = mx.dequantize(quantized_w, 
                            scales, 
                            biases, 
                            group_size=group_size, 
                            bits=bits)
print(dequantized)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/3021c24f-5c1e-4949-bcc9-dd8808791973)

**Problem2:**
When I try to adjust my group_size and bits, I set them both to 0, and when I run the code, I don't get a return value, I don't have any crash, I just keep executing, which I think is probably a bug.
```python
import mlx.core as mx

quantized_w = mx.array([[1, 2], [3, 4]],dtype=mx.uint32)
scales = mx.array([[0.5, 0.5], [0.5, 0.5]])
biases = mx.array([[1.0, 1.0], [1.0, 1.0]])

group_size = 0
bits = 0

dequantized = mx.dequantize(quantized_w, 
                            scales, 
                            biases, 
                            group_size=group_size, 
                            bits=bits)
print(dequantized)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/1f820add-8c21-474c-894e-65958d32140c)

**Problem3:**
In the mlx official documentation for this API, there is a latex text problem with the formula.
![image](https://github.com/ml-explore/mlx/assets/84373897/fda181b4-8524-4d9a-bf78-4dbcaa6fe7a9)

**Desktop (please complete the following information):**
 - OS Version:  MacOS 14.2.1
 - Version 0.10.0

**Comment by angeloskath** [comment]
Thanks for finding this.

The first one is not a bug, the error message maybe should be better but basically we require that `w` is at least divisible by the `group_size` which in this case it isn't. Also we require that `w` is actually `32 * packing_factor` but I don't think we check this during dequantize and quantize, only during matmul (I 'd have to check to make sure).

The second is obviously a bug, it doesn't make sense to have 0s in either of these values, we should throw a proper error message.

**Comment by Redmept1on** [comment]
Thanks for your reply Angelos! May I take the liberty of asking how I should modify my first piece of code to make it work? @angeloskath

**Comment by Redmept1on** [comment]
I tried to adjust my parameter input in various ways, but I still couldn't get it to work properly. 🤕

**Comment by awni** [comment]
The doc looks ok to me:

<img width="817" alt="Screenshot 2024-01-26 at 6 27 48 AM" src="https://github.com/ml-explore/mlx/assets/1542805/ee183170-ac33-4df6-bb80-906523507dcf">

**Comment by awni** [comment]
A good way to get a valid input for dequantize is to use quantize :)

```
>>> a = mx.zeros((256, 256))
>>> a,w,b = mx.quantize(a, 32, 4)
>>> mx.dequantize(a, w, b, 32, 4).shape
[256, 256]
```

**Comment by Redmept1on** [comment]
Sorry, it could be a browser problem, thanks for replying, problem2 could be listed as a bug? @awni

**Comment by awni** [comment]
Yes that is a bug, the fix is in #560
